apps/prairielearn/src/lib/client/page-context.test.ts

```diff
@@ -0,0 +1,143 @@
+  type StudentCourseInstanceContext,
+  getCourseInstanceContext,
+  getPageContext,
+} from './page-context.js';
+
+describe('getPageContext', () => {
+  it('strips extra fields from the data', () => {
+    const mockData = {
+      authz_data: {
+        has_course_instance_permission_edit: true,
+        has_course_instance_permission_view: true,
+        has_course_permission_own: true,
+        user: { name: 'Test User', uid: 'test@illinois.edu' },
+        mode: 'edit',
+      },
+      urlPrefix: '/pl/course/1/course_instance/1',
+      access_as_administrator: false,
+      news_item_notification_count: 0,
+      authn_is_administrator: false,
+      authn_user: { name: 'Test User', uid: 'test@illinois.edu' },
+      viewType: 'instructor',
+      extraField: 'this should be stripped',
+      anotherExtraField: 123,
+    };
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { extraField, anotherExtraField, ...expected } = mockData;
+
+    const result = getPageContext(mockData);
+
+    expect(result).toEqual(expected);
+  });
+
+  it('throws error when required fields are missing', () => {
+    const invalidData = {
+      // Missing most fields
+      authz_data: {
+        has_course_instance_permission_edit: true,
+      },
+    };
+
+    expect(() => getPageContext(invalidData)).toThrow();
+  });
+});
+
+describe('getCourseInstanceContext', () => {
+  const mockStudentData: StudentCourseInstanceContext = {
+    course_instance: {
+      assessments_group_by: 'Set',
+      course_id: '1',
+      deleted_at: null,
+      display_timezone: 'America/Chicago',
+      hide_in_enroll_page: false,
+      id: '1',
+      json_comment: null,
+      long_name: 'Example Student Course Instance',
+      share_source_publicly: true,
+      short_name: 'Example Student Course',
+      uuid: '1',
+    },
+    course: {
+      deleted_at: null,
+      display_timezone: 'America/Chicago',
+      id: '1',
+      short_name: 'Example Student Course',
+      announcement_color: 'red',
+      announcement_html: '<p>Hello, world!</p>',
+      course_instance_enrollment_limit: 10,
+      created_at: new Date(),
+      example_course: false,
+      institution_id: '1',
+      template_course: false,
+      title: 'Example Student Course',
+    },
+  };
+  const mockInstructorData: StaffCourseInstanceContext = {
+    course_instance: {
+      ...mockStudentData.course_instance,
+      enrollment_limit: 10,
+      sync_errors: null,
+      sync_job_sequence_id: null,
+      sync_warnings: null,
+    },
+    course: {
+      ...mockStudentData.course,
+      path: 'example/path',
+      json_comment: null,
+      sync_errors: null,
+      sync_job_sequence_id: null,
+      sync_warnings: null,
+      branch: 'main',
+      commit_hash: '1234567890',
+      repository: 'https://github.com/example/example.git',
+      sharing_name: 'example',
+      sharing_token: '1234567890',
+      show_getting_started: false,
```

Comment:
Things might get a little weird if we ever want to display lists of arbitrary courses to instructors. We probably wouldn't want to show most of these, but especially not the `sharing_token`.